include/highfive/H5DataType.hpp

```diff
@@ -80,7 +92,58 @@ class AtomicType : public DataType {
 
     typedef T basic_type;
 };
-}
+
+
+///
+/// \brief A structure representing a set of fixed-length strings
+///
+/// Although fixed-len arrays can be created 'raw' without the need for
+/// this structure, to retrieve results efficiently it must be used.
+///
+template <std::size_t N>
+class FixedLenStringArray {
+  public:
+    FixedLenStringArray() = default;
+
+    ///
+    /// \brief Create a FixedStringArray from a raw contiguous buffer
+    ///
+    FixedLenStringArray(const char array[][N], std::size_t length);
+
+    ///
+    /// \brief Create a FixedStringArray from a sequence of strings.
+    ///
+    /// Such conversion involves a copy, original vector is not modified
+    ///
+    FixedLenStringArray(const std::string* iter_begin, const std::string* iter_end);
+
+    explicit FixedLenStringArray(const std::vector<std::string> & vec);
+
+    FixedLenStringArray(const std::initializer_list<std::string> &);
+
+    ///
+    /// \brief Append an std::string to buffer structure
```

Comment:
```suggestion
    /// \brief Append a std::string to the buffer structure
```

?

Reply:
my english tells me to use `an` since we pronounce 'esse' in s-t-d 😃 but will ask Jim